pymeasure/instruments/racal/racal1992.py

```diff
@@ -0,0 +1,363 @@
+            'trigger'       : { 'auto' : 'AAU',  'manual' : 'AMN' },    # noqa
+            'impedance'     : { '50'   : 'ALI',  '1M'     : 'AHI' },    # noqa
+            'slope'         : { 'pos'  : 'APS',  'neg'    : 'ANS' },    # noqa
+            'filtering'     : { True   : 'AFE',  False    : 'AFD' },    # noqa
+            'trigger_level' : None,                                     # noqa
+            },
+        'B' : {                                                         # noqa
+            'coupling'      : { 'AC'        : 'BAC',  'DC'     : 'BDC' },    # noqa
+            'attenuation'   : { 'X1'        : 'BAD',  'X10'    : 'BAE' },    # noqa
+            'trigger'       : { 'auto'      : 'BAU',  'manual' : 'BMN' },    # noqa
+            'impedance'     : { '50'        : 'BLI',  '1M'     : 'BHI' },    # noqa
+            'slope'         : { 'pos'       : 'BPS',  'neg'    : 'BNS' },    # noqa
+            'input_select'  : { 'separate'  : 'BCS',  'common' : 'BCC' },    # noqa
+            'trigger_level' : None,                                     # noqa
+            },
+    }
+
+    operating_modes = {
+        'self_check'      : 'CK',    # noqa
+        'frequency_a'     : 'FA',    # noqa
+        'period_a'        : 'PA',    # noqa
+        'phase_a_rel_b'   : 'PH',    # noqa
+        'ratio_a_to_b'    : 'RA',    # noqa
+        'ratio_c_to_b'    : 'RC',    # noqa
+        'interval_a_to_b' : 'TI',    # noqa
+        'total_a_by_b'    : 'TA',    # noqa
+        'frequency_c'     : 'FC',    # noqa
+    }
+
+    def decode(v, allowed_types=None):
+        """Decode received message.
+
+        All values returned follow the same format: 2 letters to indicate
+        the type of the value returned, followed by a floating point number
+        (which could be an integer, of course.)
+        This here, for example, is math constant Z: MZ+001.00000000E+00
+        """
+        if len(v) != 19:
+            raise ReturnValueError("Length of instrument response must always be 19 characters")
+
+        val_type = v[0:2]
+        val = float(v[2:19])
+
+        if allowed_types and val_type not in allowed_types:
+            raise ValueError(f"Unexpected value type returned: '{val_type}'")
+
+        if val_type in Racal1992.int_types:
+            return int(val)
+        elif val_type in Racal1992.float_types:
+            return val
+        else:
+            raise ValueError("Unsupported return type")
+
+    operating_mode = Instrument.setting(
+            "%s",
+            """ Set operating mode.
+
+            Permitted modes are:
+                'self_check',
+                'frequency_a',
+                'period_a',
+                'phase_a_rel_b',
+                'ratio_a_to_b',
+                'ratio_c_to_b',
+                'interval_a_to_b',
+                'total_a_by_b',
+                'frequency_c'
+            """,
+            set_process=(lambda mode: Racal1992.operating_modes[mode]),
```

Comment:
```suggestion
            values=Racal1992.operating_modes,
            map_values=True,
```
I think this is a better fit for your needs and it should be equivalent to the `set_process` parameters.

Reply:
I don't understand how that works, but I tried your changes. It resulted in an error. 
When choosing between making this fix or not submitting this driver (I don't want to spend more time on this), I think the first choice is the better option.